addons/mail/static/src/core/common/composer.js

```diff
@@ -118,8 +116,13 @@ export class Composer extends Component {
         this.suggestion = this.store.user ? useSuggestion() : undefined;
         this.markEventHandled = markEventHandled;
         this.onDropFile = this.onDropFile.bind(this);
-        if (this.props.dropzoneRef && this.allowUpload) {
-            useDropzone(this.props.dropzoneRef, this.onDropFile, "o-mail-Composer-dropzone");
+        if (this.props.dropzoneRef) {
+            useDropzone(
+                this.props.dropzoneRef,
+                this.onDropFile,
+                "o-mail-Composer-dropzone",
+                () => this.allowUpload
+            );
```

Comment:
Not your fault, but it sucks that component hooks are not reactive

Reply:
Actually we could use `useComponent().allow_upload`, at least I can try if you think it's better